### PR TITLE
Add ocean.assets.isConsumable function

### LINKED ISSUE #759: Add ocean.assets.isConsumable
This function should check if the asset is consumable by :
- check isDisabled flag (it exists & true -> asset is not consumable)
- check file conectivity
- check file length (optional)
- check credentials

Return should be a interface, containing a status and a status_message: Eg
   0, All good
   1, Asset is disabled
   2. Asset is empty
   3, Connectivity check failure
   4, Credential missing from allow list
   5, Credential found on deny list

### PR DESCRIPTION
**Pull request by krisliew**
Fixes #759 

The function should check the data asset ddo if is consumable by:

- check isOrderDisabled flag in `status` object of ddo (if exists & true -> asset is not consumable)

Return is an interface, containing a status and a status_message: Eg
- 0, All good
- 1, Ordering this asset has been temporarily disabled by the publisher.

### CODE REVIEW

**Review by alexcos20 — CHANGES_REQUESTED** [review]
Missing:
- order should check this flag and reject if needed
- tests

**Review by alexcos20 — CHANGES_REQUESTED** [review]
Please add the following tests:
  - Alice should update her asset and set isDisable = true
  - Bob should no be able to consume Alice's dataset after that

**Comment by kremalicious** [comment]
There is currently a proposal to make `isDisabled` into `metadata.curation.isOrderDisabled`:

- https://github.com/oceanprotocol/OEPs/pull/236
- https://github.com/oceanprotocol/ocean.js/pull/789

File: src/ocean/Assets.ts
Comment (kremalicious) [inline src/ocean/Assets.ts:659]: we should be more clear what exactly is disabled, and by whom. So maybe like so:

```suggestion
        message: 'Ordering this asset has been temporarily disabled by the publisher.'
```

File: src/ocean/Assets.ts
Comment (kremalicious) [inline src/ocean/Assets.ts:653]: from the client side, for the use case for disabling the buy button when user arrives at asset details page, it would help to not fetch the DDO here again cause in almost all use case we already have the DDO. So should either accept a DDO or DID. And when DDO is passed, it should not refetch the DDO.

**Comment by krisliew** [comment]
Will need a rebase / merge against `main` after #789 is merged into `main`.

**Review by alexcos20 — APPROVED** [review]

**Comment by alexcos20** [comment]
@kremalicious - this supersedes #789, so we can close that

still blocked by stable aqua, so don't merge it yet

File: src/ddo/interfaces/Status.ts
Comment (mihaisc) [inline src/ddo/interfaces/Status.ts:11]: from the comment this looks like `isInPurgatory`, or what is the purpose? @kremalicious @alexcos20

Comment (alexcos20) [reply src/ddo/interfaces/Status.ts:11]: we decided for now that we are going to use only isOrderDisabled

**Pull request merged**
